feat(linkcard): add rich Link Card transformer & component

Tsuuuuuuun opened the pull request:
## Description
This PR introduces a **Link Card** feature that renders rich previews for bare URLs
in Markdown content.

### Key Features
- **`LinkCardHandler.tsx`** – React component to display link cards with image, title, and description.
- **Transformer `plugins/transformers/linkcard.ts`** – Converts plain external URLs into custom `<LinkCard>` nodes at build-time.
- **Inline script** – Prefetches OpenGraph data and falls back gracefully when metadata is missing.
- **SCSS styling** – Responsive, accessible design with light/dark-mode support.
- **Config updates** – Registers the transformer and mounts the component automatically.
- **Demo** – Added examples in `content/index.md`.

### Screenshots

Light

![image](https://github.com/user-attachments/assets/30dbfc50-72a2-48e6-a6ff-850a93cb2199)

Dark

![image](https://github.com/user-attachments/assets/53d48f97-8c8c-4c46-acf1-381043e33af3)


### How to Test (Installation)
1. `npm i`
2. `npm quartz create`
3. `npm i remark-link-card`
4. `npx quartz build --serve`
5. Open any page that contains a bare link

---

_Thank you for maintaining Quartz!_

Tsuuuuuuun committed:
Link Card View

Tsuuuuuuun committed:
translate comments into English

Tsuuuuuuun committed:
fix: prefix unused props in LinkCardHandler

Renamed displayClass and cfg props in LinkCardHandler to _displayClass and _cfg to explicitly indicate they are intentionally unused.

Tsuuuuuuun committed:
Run npx prettier . --write

saberzero1 commented on content/index.md:
Don't include this.

saberzero1 commented on content/.gitkeep:
Don't remove this.

saberzero1 commented on quartz/components/styles/linkcard.scss:
Why are there two `scss` files with overlapping styles?

saberzero1 commented on quartz/styles/base.scss:
```diff
@@ -3,6 +3,7 @@
 @use "./variables.scss" as *;
 @use "./syntax.scss";
 @use "./callouts.scss";
+@use "../components/styles/linkcard.scss";
```
This shouldn't be included here.

saberzero1 reviewed (commented):
Thanks for submitting this PR.

It seems you both precompute rich links and fetch rich links at runtime. This should probably one or the other.

We will likely keep this opened as a plugin until we have the community plugins system up and running.

Tsuuuuuuun committed:
re-add gitkeep to content

Tsuuuuuuun committed:
refactor(linkcard): address feedback from saberzero1

Tsuuuuuuun commented:
@saberzero1 Thanks for the valuable feedback! I’ve addressed your comments:

## Redundant files removed
- Deleted the old `quartz/styles/linkcard.scss` and consolidated all styles into `quartz/components/styles/linkcard.scss`.

## Architecture clarification
It looks like there might have been a misunderstanding around where link metadata is fetched.  
- **Build-time** (`linkcard.ts`): uses `remark-link-card` to fetch title, description, and images.  
- **Runtime** (`linkcard.inline.ts`): only hides broken images with an `error` listener (`{ once: true }`). No duplicate network requests occur.

##  Architecture polish
- Removed unused props from `LinkCardHandler`.
- Added `{ once: true }` to the image-error listener for efficiency.
